javaslang-benchmark/src/test/java/javaslang/idiom/TryBenchmark.java

```diff
@@ -0,0 +1,111 @@
+package javaslang.idiom;
+
+import javaslang.JmhRunner;
+import javaslang.collection.Array;
+import javaslang.control.Try;
+import org.junit.Test;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+
+import java.util.Random;
+
+import static javaslang.API.Case;
+import static javaslang.API.Match;
+import static javaslang.Predicates.instanceOf;
+
+/**
+ * Benchmark for Try vs try/catch.
+ */
+public class TryBenchmark {
+    static final Array<Class<?>> CLASSES = Array.of(
+            TryThrowsCatchBenchmark.class,
+            TryNoThrowsCatchBenchmark.class
+    );
+
+    @Test
+    public void testAsserts() {
+        JmhRunner.runDebugWithAsserts(CLASSES);
+    }
+
+    public static void main(String... args) {
+        JmhRunner.runNormalNoAsserts(CLASSES);
+    }
+
+
+    @State(Scope.Benchmark)
+    public static class Base {
+
+        int random;
+
+        Integer throwsException() {
+            if ((random ^ random) == 0) throw new RuntimeException();
+            return 1;
+        }
+
+        Integer doesntThrow() {
+            if ((random ^ random) != 0) throw new RuntimeException();
+            return 0;
+        }
+
+        @Setup
+        public void setup() {
+            random = new Random().nextInt();
+        }
+    }
+
+    public static class TryThrowsCatchBenchmark extends Base {
+
+        @Benchmark
+        public Object javaslang_try_actual_exception() {
+            Integer res = Try.of(this::throwsException)
+                    .recover(x -> Match(x).of(Case(instanceOf(RuntimeException.class), () -> 1)))
+                    .getOrElse(0);
+
+            assert res == 1;
+            return res;
+        }
+
+        @Benchmark
+        public Object java_try_actual_exception() {
+            Integer res = 0;
```

Comment:
you could make this a `void` and andd a `Blackhole bh` as parameter
try {
	this.throwsException();
	assert false
} catch (RuntimeException e) {
        bh.consume(e);
}

(please apply these suggestions to the rest of the codebase also)